packages/now-static-build/src/utils/gatsby.ts

```diff
@@ -0,0 +1,78 @@
+import { PackageJson } from '@vercel/build-utils';
+import { promises as fs } from 'fs';
+import * as path from 'path';
+import {
+  fileExists,
+  readPackageJson,
+  DeepWriteable,
+  writePackageJson,
+} from './_shared';
+
+const defaultConfig = {
+  plugins: [
+    {
+      resolve: 'gatsby-plugin-vercel',
+      options: {},
+    },
+  ],
+};
+
+export async function injectVercelAnalyticsPlugin(dir: string) {
+  // Gatsby requires a special variable name for environment variables to be
+  // exposed to the client-side JavaScript bundles:
+  process.env.GATSBY_VERCEL_ANALYTICS_ID = process.env.VERCEL_ANALYTICS_ID;
+
+  const gatsbyConfigName = 'gatsby-config.js';
+  const gatsbyPluginPackageName = 'gatsby-plugin-vercel';
+
+  const gatsbyConfigPath = path.join(dir, gatsbyConfigName);
+
+  const pkgJson: DeepWriteable<PackageJson> = (await readPackageJson(
+    dir
+  )) as DeepWriteable<PackageJson>;
+  if (!pkgJson.dependencies) {
+    pkgJson.dependencies = {};
+  }
+  if (!pkgJson.dependencies[gatsbyPluginPackageName]) {
+    pkgJson.dependencies[gatsbyPluginPackageName] = 'latest';
+
+    await writePackageJson(dir, pkgJson);
+  }
+
+  if (await fileExists(gatsbyConfigPath)) {
+    await fs.rename(
+      gatsbyConfigPath,
+      gatsbyConfigPath + '.__vercel_builder_backup__.js'
+    );
+
+    await fs.writeFile(
+      gatsbyConfigPath,
+      `const userConfig = require("./gatsby-config.js.__vercel_builder_backup__.js");
+
+const vercelConfig = Object.assign({}, userConfig);
```

Comment:
Looks like a shallow clone here but then `plugins` is mutated below.

What is the purpose of the clone?

Reply:
Oh I see another clone with `vercelConfig.plugins = vercelConfig.plugins.slice()` so that should work then 👍

Reply:
Plugins is sliced before we push into it. The shallow clone of both the array and full user object is to prevent mutation.

Reply:
Would be happy to tackle these nits in a follow up to not re-trigger CI 👍